### PR TITLE
#181 acknowledge modal

### LINKED ISSUE #181: acknowledge modal causing excessive re-renders
**Describe the bug**
Been reported that the acknowledge modal is sluggish

This is because it's causing excessive re-renders on the list page which is very expensive to re-render

### PR DESCRIPTION
**Pull request by joshxg**
Fixes #181

### CODE REVIEW

File: src/common/services/FormatService/FormatService.ts [hunk 1]
```diff
@@ -47,7 +47,7 @@ export class FormatService {
     return moment(date).format('DD MMMM');
   };
 
-  fileDate = (date: number): string => {
+  fullDate = (date: number): string => {
```
Comment (joshxg) [inline src/common/services/FormatService/FormatService.ts:50]: using in multiple places now so renamed it..

File: src/common/translations/en.json [hunk 2]
```diff
@@ -128,5 +128,9 @@
   "SENSORS_MAC_ADDRESS": "This sensors mac address",
   "SENSOR_DETAILS": "SENSOR DETAILS",
   "SENSORS_BATTERY_LEVEL": "This sensors battery level",
-  "REMOVE": "Remove"
+  "REMOVE": "Remove",
+  "BREACH_ONGOING": "The current breach is ongoing!",
+  "STARTED": "started",
+  "ENDED": "ended",
+  "ACKNOWLEDGE": "ACKNOWLEDGE"
```
Comment (joshxg) [inline src/common/translations/en.json:135]: translations for acknowledge modal

File: src/features/Breach/AcknowledgeBreach/AcknowledgeBreachSlice.ts [hunk 3]
```diff
@@ -1,17 +1,17 @@
...
-import { RootState } from '../../../common/store/store';
-
-interface AcknowledgeBreachSliceState {
-  acknowledgingSensorId: string;
-  acknowledging: boolean;
-  updating: boolean;
-  fetching: boolean;
-  unacknowledged: TemperatureBreach[];
-}
+import { call, put, takeEvery } from 'redux-saga/effects';
```
Comment (joshxg) [inline src/features/Breach/AcknowledgeBreach/AcknowledgeBreachSlice.ts:3]: Remove unused code, extracted types, added selectors

File: src/ui/components/modal/AcknowledgeBreachModal.tsx [hunk 4]
```diff
@@ -1,92 +1,89 @@
 import React, { FC } from 'react';
 import { useDispatch, useSelector } from 'react-redux';
-import moment from 'moment';
 import { ActivityIndicator, FlatList } from 'react-native';
 
 import { SettingsEditButtonGroup, SettingsEditModalLayout, Column, Centered } from '~layouts';
 import { t } from '~translations';
-import { AcknowledgeBreachAction } from '~features';
+import { AcknowledgeBreachAction, AcknowledgeBreachSelector } from '~features';
```
Comment (joshxg) [inline src/ui/components/modal/AcknowledgeBreachModal.tsx:7]: added selector usage, extracted flatlist render function to a componenet

**Comment by mark-prins** [comment]
I'm getting an error with non-serializable state: 

image

Should I worry about that? The value displayed looks serializable to me 🤷‍♂️

**Review by mark-prins — APPROVED** [review]
I have nothing to compare performance with unfortunately.. I didn't even know that you could long-press the breaches! 
The modal shows, and doesn't re-render... so... LGTM 😄

**Comment by joshxg** [comment]
added 
602ec04 for the unserializable state (class instances)

**Pull request merged**